feature/LP-128 Badge Button corrections/code review

JoseGR98 opened the pull request:
LP-128
https://devaptivist.atlassian.net/browse/LP-128

Within "Button.kt" file, function "BadgeButton" is now able to receive an "image vector" as parameter (icon, in case to changed).

BadgeButton with Icon A & notifications
![foto1](https://user-images.githubusercontent.com/25538206/230687356-73f03b3c-7b86-4166-b22b-c89793fc51c9.png)

BadgeButton with Icon A & no notifications
![foto2](https://user-images.githubusercontent.com/25538206/230687378-5b539428-296a-491c-9728-3d9bc96aff11.png)

BadgeButton with Icon B & notifications
![foto3](https://user-images.githubusercontent.com/25538206/230687392-d6c02a0d-b2f2-4213-a150-116351b14f73.png)

BadgeButton with Icon B & no notifications
![foto4](https://user-images.githubusercontent.com/25538206/230687405-73c5a055-ea10-4a15-9dc7-17c3e6e5eff2.png)

JoseGR98 commented:
Colors for Badge have not been created on Android project yet.
Colors marked on Jira, to be added on "Theme.kt" file within the project.
Color to fill circle --> #B3261E
Color to fill number --> #FFFFFF

Task was assigned already. Colors to be added soon. In progres.

Commit:
fun "BadgeButton" updated with latest code. Addressed Jose´s changes.

Commit:
fun "BadgeButton" refactored with name fun "LpBadgeButton"

melissaquezada94 commented on core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt:
```diff
@@ -125,45 +119,56 @@ fun LpEditFloatingActionButton(
             )
         }
-
         if (badgeNumber > 0) {
```
I think should be more flexibility this function if you send a new parameter boolean called `showBadge`, in this case the function that implements this component has the option to show the badge with the number 0, and in badgeNumber parameter set by default the number 0 like ` badgeNumber: Int = 0,`

JoseGR98 replied:
Got it. Thanks for the feedback. Already made a new commit & push with changes.

Commit:
fun LpBadgeButton upgraded based on latest comments.
Parameter "showBadge" (Boolean) was added.
Default value of "showBadge" established to "false".
Default value of "badgeNumber" established to 0.
Documentation upgraded

carlosph816 commented on core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt:
```diff
@@ -125,45 +119,58 @@ fun LpEditFloatingActionButton(
+                            .align(alignment = Alignment.TopEnd)
+                    ) {
+                        Text(
+                            text = badgeNumber.toString(),
```
What do you think if we change badgeNumber to String and we avoid toString cause it is unnecessary if it will always be cast

JoseGR98 replied:
Agree, already changed

carlosph816 commented on core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt:
```diff
@@ -125,45 +119,58 @@ fun LpEditFloatingActionButton(
-                contentDescription = "Notifications button",
-                tint = Drab
+                imageVector = imageVector,
+                contentDescription = "icon image",
```
Add this value to strings file

JoseGR98 replied:
Done

carlosph816 reviewed (commented):
Waiting changes...

carlosph816 reviewed (approved):
Well done!

Commit:
Update to fun "LpBadgeButton", based on Carlos comments.
Waiting for code review

Commit:
Merge branch 'develop' into feature/lp-128-BadgeButton-Icon-fix

# Conflicts:
#	core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt

Commit:
"Button.kt"- LpBadgeButton updated with Carlos comments.
Only fun LpBadgeButton modified.
Added standard padding measures of LeaspertTheme.
